docs/reference/technology/connectivity/cellular.md

```diff
@@ -0,0 +1,159 @@
+
+This feature offers great power savings for periodically reporting devices. Between reports, the device is in PSM state with the modem in the deep sleep state. The latency of contacting the device is the PSM period.
+
+PSM configuration is negotiated with the network, and the actual PSM time that network has accepted may differ from that requested.
+
+#### eDRX - extended Discontinuous Reception
+
+    opt_receive_period(int mode, EDRXAccessTechnology act_type, uint8_t edrx_value)
+
+eDRX tells how long the device sleeps in continuous connection. The device needs to be able to receive data, but it can tell the network that it checks for incoming messages, for example only every 200 seconds. It can receive messages but only at the given times to allow battery saving sleep periods. These time values are greatly extended compared to normal 4G data transmission – hence the name.
+
+This feature serves devices that need smaller latencies. A connection is kept open all the time, but the modem achieves a sleep state between the reception times. The data connection, if using IPv4, may need periodic keep alive messages to keep the network address translation mapping valid.
+
+eDRX configuration is given to the modem. It is negotiated with the network, and the time accepted by the network may differ from the requested time. Availability of this optimization depends on the cellular network.
+
+#### Sleep more, save energy
+
+CellularConnectionFSM simplifies connecting to a cellular network. `CellularConnectionFSM` is a reliable way to connect, but you may want to optimize it further when implementing applications for constrained battery-operated devices. Figure 5 illustrates the `CellularConnectionFSM` operation.
+
+<span class="images">![](https://s3-us-west-2.amazonaws.com/mbed-os-docs-images/api-cellular-connect.png)<span>Figure 5. Cellular connection process in general</span></span>
+
+### Considerations for UDP, TCP and non-IP use
+
+Which networking protocol to use depends on multiple factors. Server communication model, power consumption, reliability need and operator support are the biggest factors. 
+
+TCP is a reliable transmission protocol. For long-lived sessions, periodic keep alive messaging is necessary, which places demands on the main power source. For a long-lived TCP connection, the server application can contact the device. Connections can still have long latencies because the device's use of eDRX power optimization affects when it listens for incoming packets. Mbed TLS supports TLS transport security over TCP. Operators may, however, prevent using TCP over NB-IoT due to device deployment and network planning considerations.
```

Comment:
Query: Do we need a TLS representative to review this content?

Reply:
I think this is good as it is, the chapter was originally written by @lauri-piikivi

Reply:
Would be nice to point at TLS docs. Maybe this as entry point? 
https://tls.mbed.org/core-features

Reply:
Added link to TLS, because this is for app devs code example link serves here the best
https://github.com/ARMmbed/Handbook/pull/489/commits/3d1cb4bcf7af7d7f4a17b191550ca29ca79a7e0d